kythe/go/platform/analysis/proxy/proxy.go

```diff
@@ -110,13 +122,20 @@ type response struct {
 	Args   interface{} `json:"args,omitempty"`
 }
 
-// A unit represents a compilation unit to be analyzed.
+// A unit represents a compilation unit (in JSON form) to be analyzed.
 type unit struct {
 	Unit            json.RawMessage `json:"unit"`
 	Revision        string          `json:"rev,omitempty"`
 	FileDataService string          `json:"fds,omitempty"`
 }
 
+// A unit represents a compilation unit (in base64-encoded wire format) to be analyzed.
+type unit_wire struct {
+	Unit            string `json:"unit"`
```

Comment:
Rather than making this a `json.RawMessage` and explicitly encoding, I recommend making this field a plain `[]byte`. The JSON encoder will already encode byte slices as a (JSON) string encoded in Base64 (see https://pkg.go.dev/encoding/json).

Reply:
Thanks for pointing this out! I have updated the PR based on your suggestion